include/libopencm3/stm32/h7/pwr.h

```diff
@@ -57,39 +61,133 @@ LGPL License Terms @ref lgpl_license
 /** DBP[8]: Disable backup domain write protection. */
-#define PWR_CR1_DBP               (1 << 8)
+#define PWR_CR1_DBP               BIT8
 
 /** CSR1 Register Bits */
+#define PWR_CSR1_MMCVDO           BIT17
 #define PWR_CSR1_AVDO             BIT16
 #define PWR_CSR1_ACTVOS_SHIFT     14
 #define PWR_CSR1_ACTVOSRDY        BIT13
 #define PWR_CSR1_PVDO             BIT4
 
+/** CR2 Register Bits */
+/** temperature level monitoring versus high threshold */
+#define PWR_CR2_TEMPH             BIT23
+/** temperature level monitoring versus low threshold */
+#define PWR_CR2_TEMPL             BIT22
+/** backup regulator ready */
+#define PWR_CR2_BRRDY             BIT16
+/** V BAT and temperature monitoring enable */
+#define PWR_CR2_MONEN             BIT4
+/** backup regulator enable */
+#define PWR_CR2_BREN              BIT0
+
 /** CR3 Register Bits */
 #define PWR_CR3_USB33RDY          BIT26
 #define PWR_CR3_USBREGEN          BIT25
 #define PWR_CR3_USB33DEN          BIT24
+/** SMPS step-down converter external supply ready */
+#define PWR_CR3_SMPSEXTRDY        BIT16
+/** V BAT charging resistor selection  */
 #define PWR_CR3_VBRS              BIT9
+/** V BAT charging enable */
 #define PWR_CR3_VBE               BIT8
+
+/** SMPS step-down converter voltage output level selection
+ *  This setting is used when both the LDO and SMPS step-down converter are enabled with SMPSEN and
+ *  LDOEN enabled or when SMPSEXTHP is enabled. In this case SMPSLEVEL must be written with
+ *  a value different than 00 to reach the appropriate voltage, based on VOS or external supply.
+ */
+#define PWR_CR3_SMPSLEVEL_SHIFT   4
+#define PWR_CR3_SMPSLEVEL_MASK    0x3
+#define PWR_CR3_SMPSLEVEL_VOS     0x0
+#define PWR_CR3_SMPSLEVEL_1P8V    0x1
+#define PWR_CR3_SMPSLEVEL_2P5V    0x2
+
+/** SMPS step-down converter external power delivery selection */
+#define PWR_CR3_SMPSEXTHP         BIT3
 #define PWR_CR3_SCUEN             BIT2
+/* BIT2 Is overloaded on devices with SMPS as the SMPSEN bit. */
+#define PWR_CR3_SMPSEN            BIT2
 #define PWR_CR3_LDOEN             BIT1
 #define PWR_CR3_BYPASS            BIT0
 
 /** D3CR Register Bits */
-#define PWR_D3CR_VOS_SHIFT        14
 #define PWR_D3CR_VOSRDY           BIT13
-
+#define PWR_D3CR_VOS_SHIFT        14
+#define PWR_D3CR_VOS_MASK         (0x03)
+/** VOS0 is implemented on STM32H72x/3x with simple VOS setting.
+ *  STM32H742/43/45/47/50/53/55/57 SCALE0 this as SCALE1 + SYSCFG.ODEN */
+#define PWR_D3CR_VOS_SCALE_0      (0x0)
 #define PWR_D3CR_VOS_SCALE_3      (0x1)
 #define PWR_D3CR_VOS_SCALE_2      (0x2)
 #define PWR_D3CR_VOS_SCALE_1      (0x3)
-#define PWR_D3CR_VOS_MASK         (0x03)
+/** SRDCR Register Bits */
+#define PWR_SRDCR_VOSRDY          BIT13
+#define PWR_SRDCR_VOS_SHIFT       14
+#define PWR_SRDCR_VOS_MASK        (0x03)
+#define PWR_SRDCR_VOS_SCALE_3     (0x0)
+#define PWR_SRDCR_VOS_SCALE_2     (0x1)
+#define PWR_SRDCR_VOS_SCALE_1     (0x2)
+#define PWR_SRDCR_VOS_SCALE_0     (0x3)
```

Comment:
oh that's gross, boo ST! :)

Reply:
Tell me about it!  Also, fun debugging this stuff because the LDO/SMPS/etc settings (not including VOS) can only be written once per power cycle :-)